src/oop/special_class_method/hard/q13/MainTest.java

```diff
@@ -0,0 +1,156 @@
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static global.tools.CustomAssertions._assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class MainTest {
+    // Parsons with Distractors
+
+    private final String className = "Cup";
+    private final String attributeName1 = "colour";
+    private final String attributeName2 = "volume";
+    private final String attributeName3 = "liquid";
+    private final String getAttributeMethodName2 = "getVolume";
+    private final String setAttributeMethodName3 = "setLiquid";
+    private ObjectTest testClass;
+
+    @BeforeEach
+    public void setUp() throws Throwable {
+        String classString = "oop.special_class_method.hard.q13." + className;
+        this.testClass = new ObjectTest(classString);
+    }
+
+    @Test
+    public void cupClassHasCorrectAttributes() {
+        assertTrue(testClass.hasField(attributeName1),
+                "Your " + className + " class is missing the " + attributeName1 + " attribute, or it is spelled incorrectly.");
+        assertTrue(testClass.hasField(attributeName1, String.class),
+                "Your " + attributeName1 + " does not have the correct type.");
+        assertTrue(testClass.hasModifier(attributeName1, "private"),
+                "Your " + attributeName1 + " does not have the correct visibility modifier.");
+        assertTrue(testClass.hasField(attributeName2),
+                "Your " + className + " class is missing the " + attributeName2 + " attribute, or it is spelled incorrectly..");
+        assertTrue(testClass.hasField(attributeName2, double.class),
+                "Your " + attributeName2 + " does not have the correct type.");
+        assertTrue(testClass.hasModifier(attributeName2, "private"),
+                "Your " + attributeName2 + " does not have the correct visibility modifier.");
+        assertTrue(testClass.hasField(attributeName3),
+                "Your " + className + " class is missing the " + attributeName3 + " attribute, or it is spelled incorrectly.");
+        assertTrue(testClass.hasField(attributeName3, String.class),
+                "Your " + attributeName3 + " does not have the correct type.");
+        assertTrue(testClass.hasModifier(attributeName3, "private"),
+                "Your " + attributeName3 + " does not have the correct visibility modifier.");
+    }
+
+
+    private static Stream<Arguments> inputProvider() {
+        return Stream.of(
+                Arguments.of("Red", 3423.4, "Milk", "Water"),
+                Arguments.of("Black", 54.5, "Water", "Lava"),
+                Arguments.of("Blue", 934.3, "Mercury", "Juice"),
+                Arguments.of("White", 49374.43, "Blood", "Water"),
+                Arguments.of("Yellow", 434.43, "Lava", "Whiskey")
```

Comment:
Add a constructor with empty strings and 0, please.

Reply:
You should also make it so the set method sets to an empty string from a non-empty one and vice versa.